scripts/eurostat/health_determinants/common/denominator_mcf_generator.py

```diff
@@ -0,0 +1,210 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file takes the generated MCF file and makes a new MCF file which would have
+Stat Vars for the measurement denominator property values.
+"""
+import re
+import os
+import sys
+
+# pylint: disable = wrong-import-position
+_COMMON_PATH = os.path.abspath(os.path.dirname(__file__))
+sys.path.insert(1, _COMMON_PATH)
+from dcid_existance import check_dcid_existance
+# pylint: enable = wrong-import-position
+
+_MODULE_DIR = os.path.join(_COMMON_PATH, "..", "bmi")
+
+_INPUT_MCF_FILE_PATH = os.path.join(_MODULE_DIR, "output_files",
+                                    "eurostat_population_bmi.mcf")
+_OUTPUT_MCF_FILE_PATH = os.path.join(_MODULE_DIR, "output_files",
+                                     "eurostat_population_bmi_deno.mcf")
+
+_INCOME_QUINTILE_VALUES = {
+    "IncomeOf0To20Percentile": "[0 20 Percentile]",
+    "IncomeOf20To40Percentile": "[20 40 Percentile]",
+    "IncomeOf40To60Percentile": "[40 60 Percentile]",
+    "IncomeOf60To80Percentile": "[60 80 Percentile]",
+    "IncomeOf80To100Percentile": "[80 100 Percentile]",
+}
+
+_NAME_PROP_REPLACEMENTS = {
+    ", Among": " Among",
+    "Population:,": "Population:",
+    "Percentage,": "Percentage",
+    "Among,": "Among",
+    "ACitizen": "A Citizen",
+    "To": "To ",
+    "Of": "Of ",
+    "To299": "To 299",
+    "To149": "To 149",
+    "Least30": "Least 30",
+    "Normalweight": "Normal Weight"
+}
+
+_ADDITIONAL_NAME_PROP_REPLACEMENTS = {' Among': ''}
+
+
+def _update_name_property(name_prop: str) -> str:
+    """
+    Updates the name property.
+    Args:
+        name_prop (str): name property
+    Returns:
+        str: updated name property
+    """
+
+    name_prop = name_prop.rstrip(', ')
+    name_prop = name_prop.rstrip('with')
+    # Adding spaces before every capital letter,
+    # to make SV look more like a name.
+    name_prop = re.sub(r"(\w)([A-Z])", r"\1 \2", name_prop)
+    name_prop = "name: \"" + name_prop + "\""
+    for key, value in _NAME_PROP_REPLACEMENTS.items():
+        name_prop = name_prop.replace(key, value)
+    if name_prop.strip().endswith('Among"'):
+        for key, value in _ADDITIONAL_NAME_PROP_REPLACEMENTS.items():
+            name_prop = name_prop.replace(key, value)
+    return name_prop
+
+
+# pylint: disable = too-many-return-statements
+# pylint: disable = too-many-branches
+def _generate_pv_node(prop: str) -> str:
+    """
+    Returns the PV for the sv property.
+    Args:
+        prop (str): property
+    Returns:
+        str: property-value
+    """
+
+    if prop in ["Count"]:
+        return "continue"
+    if "PhysicalActivity" in prop:
```

Comment:
If `in prop` is a  substring search, but the value assigned is the prop itself, then can use  adict instead of a series of if-else:
```
_VALUE_SUBSTR_TO_PROP_MAP = {
  "PhysicalActivity": "healthBehavior",
  "Education" : "educationalAttainment"
...
}
for substr, property_name in _VALUE_SUBSTR_TO_PROP_MAP:
  if substr in prop:
     return f'property_name: dcs:{prop}'
```

Can have if-else just for the more complex ones such as Activity, Mintues, etc